File: TwoPointers/ValidPalindrome/ValidPalindrome.py
class Solution(object):
    def isPalindrome(self, s):
        if not s:
            return False
        cleaned_string = self.remove_non_alpha_num_chars(s.lower())
        print(cleaned_string)
        l, r = 0, len(cleaned_string) - 1
        while l < r:
            if cleaned_string[l] != cleaned_string[r]:
                return False
            l += 1
            r -= 1
        return True
    

    def remove_non_alpha_num_chars(self, s):
        res = []
        for c in s:
            if (ord(c) >= ord('a') and ord(c) <= ord('z')) or (ord(c) >= ord('0') and ord(c) <= ord('9')):
                res.append(c)
        return ''.join(res)

File: TwoPointers/ValidPalindrome/ValidPalindrome.go
import (
	"strings"
)

func isPalindrome(s string) bool {
	cleanedString := cleanString(s)
	left, right := 0, len(cleanedString)-1
	for left < right {
		if cleanedString[left] != cleanedString[right] {
			return false
		}
		left++
		right--
	}
	return true
}

func cleanString(s string) string {
	res := strings.ToLower(s)
	var result strings.Builder
	for i := 0; i < len(res); i++ {
		b := res[i]
		if ('a' <= b && b <= 'z') ||
			('A' <= b && b <= 'Z') ||
			('0' <= b && b <= '9') {
			result.WriteByte(b)
		}
	}
	return result.String()
}